src/components/VDataTable/mixins/head.js

```diff
@@ -58,7 +58,7 @@ export default {
     genHeaderData (header, children) {
       const classes = ['column']
       const data = {
-        key: header[this.headerText],
+        key: header['key'] ? header['key'] : header[this.headerText],
```

Comment:
I'd rather see it as just `key: header[this.headerKey]` with new `headerKey` prop defaulted to `"text"`

Reply:
Shouldn't it use `value`?

Reply:
`value` could be duplicated too (if someone wants to display same column twice, for example as a first one and last one)

Reply:
Yes, that's exactly my use case. I am not that deep in the internals of vuetify yet, but as far as I understand, `key` simply needs to be something unique here. Generating it somehow would of course be preferable to passing somehting manually. Can we maybe access the column index somehow to append it to the `key`?

Reply:
Maybe pass the index down from https://github.com/vuetifyjs/vuetify/blob/23888743866cf7c74568f3810d937bc99fbe8b10/src/components/VDataTable/mixins/head.js#L29?

Reply:
Index could be a solution as well, you just need to pass it in `genTHead()` and maybe set the default value of `headerKey` to be `null` (that means index should be used)
Something like
`const row = this.headers.map((o, i) => this.genHeader(o, this.headerKey ? o[this.headerKey] : i))`

Reply:
Do we need the `headerKey` prop at all then? I mean, could there be an use case where someone would want to pass a key explicitly? Again, I am not really into the internals and think that the `key` is only there for being unique. So if this is guaranteed, there would be no need to override it.

Reply:
You may want to fully rerender your header cell when the headers are dynamically updated. Sometimes it can produce a bug if it uses index as a key, see for example https://github.com/vuetifyjs/vuetify/issues/3967